crates/chia-consensus/src/gen/solution_generator.rs

```diff
@@ -39,6 +40,48 @@ where
     Ok(quote)
 }
 
+fn clvm_bytes_len(val: u64) -> usize {
+    if val < 0x80 {
+        1
+    } else if val < 0x8000 {
+        3
+    } else if val < 0x0080_0000 {
+        4
+    } else if val < 0x8000_0000 {
+        5
+    } else if val < 0x0080_0000_0000 {
+        6
+    } else if val < 0x8000_0000_0000 {
+        7
+    } else if val < 0x0080_0000_0000_0000 {
+        8
+    } else if val < 0x8000_0000_0000_0000 {
+        9
+    } else {
+        10
+    }
+}
+
+// calculate the size in bytes of a generator with no backref optimisations
+pub fn calculate_generator_length<I>(spends: I) -> usize
+where
+    I: AsRef<[CoinSpend]>,
+{
+    let mut size: usize = 5; // (q . (())) => ff01ff8080 => 5 bytes
+
+    for s in spends.as_ref() {
+        let puzzle = s.puzzle_reveal.as_ref();
+        let solution = s.solution.as_ref();
+        // parent-id puzzle-reveal amount solution
+        // parent-id is always 32 bytes + prepend 1 byte = 33
+        // + 6 bytes for list extension
+        // coin amount is already prepended correctly in clvm_bytes_len()
+        size += 39 + puzzle.len() + clvm_bytes_len(s.coin.amount) + solution.len();
+    }
+
+    size
+}
```

Comment:
Perhaps a comment similar to https://github.com/Chia-Network/chia_rs/blob/38579c836ea560ae74b4821ce397c0d88630bdeb/crates/chia-consensus/src/gen/solution_generator.rs#L12-L14 would make all the other comments below it clearer?